src/libcollections/string.rs

```diff
@@ -765,6 +766,25 @@ impl<'a> Extend<&'a str> for String {
     }
 }
 
+// Convenience method so that you can pass in a `&String` as a pattern.
```

Comment:
This could also be a doc comment explaining that it delegates to the `&str` impl.